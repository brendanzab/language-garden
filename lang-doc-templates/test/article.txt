${let place := "world"}

${heading1 "Document template language"}

${para "
  Hello ${place}!
"}

${para "
  This is a document template language that elaborates to a typed lambda
  calculus.
"}

${heading2 "Thoughts"}

${para "
  Designing a nice concrete syntax is hard. Trying to make it not too ad-hoc and
  hard to parse, and extensible over time, while remaining lightweight and
  pleasant to use.
"}
