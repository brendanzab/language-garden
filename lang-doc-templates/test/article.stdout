

<h1>Document template language</h1>

<p>
  Hello ${place}!
</p>

<p>
  This is a document template language that elaborates to a typed lambda
  calculus.
</p>

<h2>Thoughts</h2>

<p>
  Designing a nice concrete syntax is hard. Trying to make it not too ad-hoc and
  hard to parse, and extensible over time, while remaining lightweight and
  pleasant to use.
</p>
